monai/transforms/io/array.py

```diff
@@ -227,125 +227,152 @@ def __call__(self, filename: Union[Sequence[PathLike], PathLike], reader: Option
+        output_format: an optional string of filename extension to specify the output image writer.
+            see also: `monai.data.image_writer.SUPPORTED_WRITERS`.
+        writer: a customised image writer to save data arrays.
+            if `None`, use the default writer from `monai.data.image_writer` according to `output_ext`.
     """
 
     def __init__(
         self,
         output_dir: PathLike = "./",
         output_postfix: str = "trans",
         output_ext: str = ".nii.gz",
+        output_dtype: DtypeLike = np.float32,
         resample: bool = True,
         mode: Union[GridSampleMode, InterpolateMode, str] = "nearest",
         padding_mode: Union[GridSamplePadMode, str] = GridSamplePadMode.BORDER,
         scale: Optional[int] = None,
         dtype: DtypeLike = np.float64,
-        output_dtype: DtypeLike = np.float32,
         squeeze_end_dims: bool = True,
         data_root_dir: PathLike = "",
         separate_folder: bool = True,
         print_log: bool = True,
+        output_format: str = "",
+        writer: Optional[image_writer.ImageWriter] = None,
     ) -> None:
-        self.saver: Union[NiftiSaver, PNGSaver]
-        if output_ext in {".nii.gz", ".nii"}:
-            self.saver = NiftiSaver(
-                output_dir=output_dir,
-                output_postfix=output_postfix,
-                output_ext=output_ext,
-                resample=resample,
-                mode=GridSampleMode(mode),
-                padding_mode=padding_mode,
-                dtype=dtype,
-                output_dtype=output_dtype,
-                squeeze_end_dims=squeeze_end_dims,
-                data_root_dir=data_root_dir,
-                separate_folder=separate_folder,
-                print_log=print_log,
-            )
-        elif output_ext == ".png":
-            self.saver = PNGSaver(
-                output_dir=output_dir,
-                output_postfix=output_postfix,
-                output_ext=output_ext,
-                resample=resample,
-                mode=InterpolateMode(mode),
-                scale=scale,
-                data_root_dir=data_root_dir,
-                separate_folder=separate_folder,
-                print_log=print_log,
-            )
-        else:
-            raise ValueError(f"unsupported output extension: {output_ext}.")
+        self.folder_layout = FolderLayout(
+            output_dir=output_dir,
+            postfix=output_postfix,
+            extension=output_ext,
+            parent=separate_folder,
+            makedirs=True,
+            data_root_dir=data_root_dir,
+        )
+
+        self.output_ext = output_ext.lower() or output_format.lower()
+        self.writers = image_writer.resolve_writer(self.output_ext) if writer is None else (writer,)
+        self.writer_obj = None
+
+        _output_dtype = output_dtype
+        if self.output_ext == ".png" and _output_dtype not in (np.uint8, np.uint16):
```

Comment:
Maybe we should unify the `ext` and `.XXX` with readers `suffixes` and `XXX`?

Thanks.